wallet-new/integration/Functions.hs

```diff
@@ -35,454 +37,455 @@ import qualified Pos.Wallet.Web.ClientTypes.Types as V0
+            -- We choose from the existing wallets.
+            wallet  <-  pickRandomElement localWallets
 
-    -- Modify wallet state accordingly.
-    pure $ ws
-        & wallets    .~ delete wallet localWallets
-        & actionsNum +~ 1
+            log $ "Deleting: " <> show (walId wallet)
+            -- If we don't have any http client errors, the delete was a success.
+            _       <-  either throwM pure =<< deleteWallet wc (walId wallet)
 
-runAction wc ws UpdateWallet = do
+            -- Just in case, let's check if it's still there.
+            result  <-  respToRes $ getWallets wc
 
-    let localWallets = ws ^. wallets
+            checkInvariant
+                (all ((/=) wallet) localWallets)
+                (LocalWalletsDiffers result localWallets)
 
-    -- The precondition is that we need to have wallets.
-    guard (not (null localWallets))
+            -- Modify wallet state accordingly.
+            wallets    .= delete wallet localWallets
 
-    -- We choose from the existing wallets.
-    wallet  <-  pickRandomElement localWallets
+        UpdateWallet -> do
 
-    let walletId = walId wallet
+            localWallets <- use wallets
 
-    let newWallet =
-            WalletUpdate
-                { uwalAssuranceLevel = NormalAssurance
-                , uwalName           = "Wallet name " <> (show walletId)
-                }
+            -- The precondition is that we need to have wallets.
+            guard (not (null localWallets))
 
-    result  <-  respToRes $ updateWallet wc walletId newWallet
+            -- We choose from the existing wallets.
+            wallet  <-  pickRandomElement localWallets
 
-    -- Modify wallet state accordingly.
-    pure $ ws
-        & wallets . each . filtered (== wallet) .~ result
-        & actionsNum +~ 1
+            let walletId = walId wallet
 
-runAction wc ws UpdateWalletPass = do
+            let newWallet =
+                    WalletUpdate
+                        { uwalAssuranceLevel = NormalAssurance
+                        , uwalName           = "Wallet name " <> (show walletId)
+                        }
 
-    let localWallets = ws ^. wallets
+            log $ "Updating wallet: " <> show walletId <> ", " <> show newWallet
+            result  <-  respToRes $ updateWallet wc walletId newWallet
 
-    -- The precondition is that we need to have wallets.
-    guard (not (null localWallets))
+            -- Modify wallet state accordingly.
+            wallets . each . filtered (== wallet) .= result
 
-    -- We choose from the existing wallets.
-    wallet  <-  pickRandomElement localWallets
+        UpdateWalletPass -> do
 
-    let walletId = walId wallet
+            localWallets <- use wallets
 
-    -- Get the old wallet password.
-    let oldWalletPass :: Maybe SpendingPassword
-        oldWalletPass = ws ^. walletsPass . at wallet
+            -- The precondition is that we need to have wallets.
+            guard (not (null localWallets))
```

Comment:
probably not needed - `pickRandomElement` bellow will be the guard. 

It is used in many places so won't mention it again